### ISSUE TITLE
[SCM-673] IndexOutOfBoundsException while checkouting project with recent plexus-utils

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#878]
**[Sergey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sotinov)** opened **[SCM-673](https://issues.apache.org/jira/browse/SCM-673?redirect=false)** and commented

`IndexOutOfBoundsException` exception could be thrown when using recent versions of plexus-utils.
[ERROR] Failed to execute goal XXX:YYY:1.1-SNAPSHOT:release (default-cli) on project ZZZ: 	at org.apache.maven.shared.release.phase.CheckoutProjectFromScm.execute(CheckoutProjectFromScm.java:155)
[ERROR] at org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:346)
[ERROR] at org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:293)
[ERROR] at org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:272)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:597)
[ERROR]
[ERROR] org.apache.maven.shared.release.ReleaseExecutionException: org.apache.maven.shared.release.ReleaseExecutionException: An error is occurred in the checkout process: Exception while executing SCM command.
[ERROR] -> [Help 1]

---
No further details from [SCM-673](https://issues.apache.org/jira/browse/SCM-673?redirect=false)

**Comment by jira-importer** [comment]
**[Sergey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sotinov)** commented

Exeception above caused by `IndexOutOfBoundsException` in method `parseLine` of `SvnCheckOutConsumer` class.
Here diff to fix it:

**Comment by jira-importer** [comment]
**[Sergey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sotinov)** commented

```
$ svn diff
Index: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/main/java/org/apache/maven/scm/provider/svn/svnexe/command/checkout/SvnCheckOutConsumer.java
===================================================================
--- maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/main/java/org/apache/maven/scm/provider/svn/svnexe/command/checkout/SvnCheckOutConsumer.java    (revision 1333449)
+++ maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svnexe/src/main/java/org/apache/maven/scm/provider/svn/svnexe/command/checkout/SvnCheckOutConsumer.java    (working copy)
@@ -57,7 +57,14 @@
         //[SCM-368]
         if ( file.startsWith( getWorkingDirectory().getAbsolutePath() ) )
         {
-            file = StringUtils.substring( file, getWorkingDirectory().getAbsolutePath().length() + 1 );
+           if ( file.length() > getWorkingDirectory().getAbsolutePath().length() )
+           {
+               file = StringUtils.substring( file, getWorkingDirectory().getAbsolutePath().length() + 1 );
+           }
+           else
+           {
+               file = "."
+           }
         }

         ScmFileStatus status;
```

**Comment by jira-importer** [comment]
**[Sergey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sotinov)** commented

mm, posted diff from `trunk` which is  `StringUtils`'s substring. Guess it fixes the issue.
Affected version is maven-scm-1.6

**Comment by jira-importer** [comment]
**[Sergey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sotinov)** commented

found it already fixed in 1.7
